### DISCUSSION TITLE
Docs vs implementation: should `autoForward` also forward events to other services than just machines?

### DISCUSSION THREAD

**Discussion by phryneas** [statelyai/xstate#2321] [Q&A]
**Description**

The documentation states:

> autoForward - (optional) true if all events sent to this machine should also be sent (or forwarded) to the invoked child (false by default)

This wording implies that all children, also callbacks, should get those forwarded events. In reality, those are only forwarded to machines as the actor is only added to `forwardTo` in `spawnMachine`:

https://github.com/davidkpiano/xstate/blob/master/packages/core/src/interpreter.ts#L986-L988

This has cost me about 6 hours of wondering why my code doesn't work, pulling my hair and finally stepping through the code & reading the xstate source.

I'm not sure what the right behaviour would be, so this is either a bug report against the code or the documentation.

Personally, I'd love this to be possible.

**Comment by dmitrii-esin** [comment]
Is it relevant?

**Comment by davidkpiano** [comment]
> This has cost me about 6 hours of wondering why my code doesn't work

Sorry about that; in the future, don't hesitate to go in the chat https://gitter.im/statecharts/statecharts and ask for help.

Yes, this currently only works for machines, but we can make an enhancement to make it work for callbacks too.

**Comment by phryneas** [comment]
Yeah... sorry for my tone. I was quite frustrated at that point. 
Oh, and I was in the chat, but as it is with chats: some time everyone is asleep ;)
I just lucked out that day ^^

Personally, I'm not using this at the moment, but it would still very cool if you could add it :)
